### ISSUE TITLE
Final clean up for `kedro pipeline create` and `kedro catalog create` docs

### ISSUE THREAD

**Issue by noklam** [kedro-org/kedro#2918]
## Description

Steps to reprdouce:
```
kedro new -s spaceflights --checkout main 
kedro pipeline create new
```

## Inconsistency 1 - `kedro pipeline create` add `README.md` while starter doesn't have it
![image](https://github.com/kedro-org/kedro/assets/18221871/f8b75c44-4613-4769-8f06-5a75f0e633c7)

## Inconsistency 2 - CLI docstring seems to be outdated.

https://github.com/kedro-org/kedro/blob/6888001c6019059ae717c99fbe26a06f67d73ca2/kedro/framework/cli/catalog.py#L131C1-L141C8

## Inconsistency 3 - broken doc
It generates a link that doesn't exist - https://docs.kedro.org/en/0.18.12/kedro_project_setup/configuration.html#parameters

See this:
https://github.com/kedro-org/kedro/blob/6888001c6019059ae717c99fbe26a06f67d73ca2/kedro/templates/pipeline/%7B%7B%20cookiecutter.pipeline_name%20%7D%7D/config/parameters_%7B%7B%20cookiecutter.pipeline_name%20%7D%7D.yml#L5

Also, any idea it is in a folder called `config` but not `conf`?
## Outdated Doc 4
See https://github.com/kedro-org/kedro/pull/2888#discussion_r1297170881

**Comment by amandakys** [comment]
just to confirm - this `kedro pipeline create` doesn't create the catalog_{pipeline}.yml file right? 
I think unless you think there's a good reason to add the readme.md we can just remove it? 
I think there's also another ticket tracking the update of the docs after this work.

**Comment by DimedS** [comment]
Good remarks. Perhaps we should enhance the README.md in the starters pipelines with a good examples. 
It seems the link should be updated to https://docs.kedro.org/en/0.18.12/configuration/parameters.html.
Also this would be a good place to add e2e test for 'kedro pipeline delete' that we previously discussed.

**Comment by noklam** [comment]
> just to confirm - this `kedro pipeline create` doesn't create the catalog_{pipeline}.yml file right? I think unless you think there's a good reason to add the readme.md we can just remove it? I think there's also another ticket tracking the update of the docs after this work.

@amandakys - it doesn't create `catalog_pipeline.yml`, I think the current behavior is correct and consistent with the outcome of technical design. `catalog_pipeline.yml` only created by `kedro catalog create`.

On the other hand, I am in favor of deleting it as it doesn't add much value. Some users, especially for the larger project they do document pipelines separately with `README.md`. This should be quite easy to do themselves and I don't think it's necessary for us to add it in `kedro pipeline create`.

**Comment by merelcht** [comment]
Inconsistency 1: discussed in backlog grooming all good to remove the pipeline level `README.md`.
Inconsistency 2: Change to: 
```
The catalog configuration will be saved to
    `<conf_source>/<env>/catalog_<pipeline_name>.yml` file.
```
Inconsistency 3: link should be changed to `# Link: https://docs.kedro.org/en/{{ cookiecutter.kedro_version }}/configuration.html#parameters`

**Comment by stichbury** [comment]
Following some discussion in #2888 which updates Data Catalog docs, we agreed that we need to update the text that mentions this command in the two places it appears (data catalog docs and CLI reference). Please consider those changes part of the definition of done for this issue.

**Comment by noklam** [comment]
@stichbury I actually just look into the PR again. It more likely we should fix it in #2888. It's likely due to when you start this docs change, the PR hasn't been merged so you have a older copy.

Because your change actually deleted the whole `data_catalog.md`, so it doesn't have any conflict on your PR. This PR is small and likely to get merged within a day or two.

I
